src/lib/peer_trust/peer_trust.mli

```diff
@@ -0,0 +1,82 @@
+(** Module handling peer "trust" scores. Trust scores are bounded between -1 and
+    +1, increase when peers do good things, and decrease when they do bad
+    things. If a peer's trust becomes <= -1, or takes an action that causes an
+    instant ban, it gets banned. This module tracks, stores and updates trust,
+    and notifies the rest of the system when a peer is banned. It does *not*
+    specify specific things that cause trust to change, or how much, or actually
+    do anything when a peer should be banned. That's the responsibility of the
+    caller, which is coda_base. *)
+
+open Core
+
+(** What we do in response to some trust-affecting action. *)
+module Trust_response : sig
+  type t =
+    | Insta_ban  (** Ban the peer immediately *)
+    | Trust_change of float
+        (** Change trust by the specified amount, positive or negative. *)
+end
+
+(** Interface for trust-affecting actions. *)
+module type Action_intf = sig
+  (* TODO add a show interface for logging. *)
+  type t
+
+  val to_trust_response : t -> Trust_response.t
+end
+
+val max_rate : float -> float
+(** Trust increment that sets a maximum rate of doing a bad thing (presuming the
```

Comment:
There's an exponent so no. (The definition is `((Record.decay_rate ** secs) *. -1.) +. 1.`. As written it's the number of seconds between doing the thing, but it's probably clearer to flip it around (reciprocate), especially since I named it "rate".